phosphor-auditlog/auditLogUtils.hpp

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include <libaudit.h>
+
+#include <phosphor-logging/lg2.hpp>
+#include <xyz/openbmc_project/Common/File/error.hpp>
+
+#include <filesystem>
+#include <fstream>
+#include <string>
+
+namespace phosphor
+{
+namespace auditlog
+{
+
+/** @class ALParseFile
+ *  @brief Creation of temporary file for parsed audit log
+ */
+class ALParseFile
+{
+  public:
+    ALParseFile(const ALParseFile&) = delete;
+    ALParseFile& operator=(const ALParseFile&) = delete;
+    ALParseFile(ALParseFile&&) = delete;
+    ALParseFile& operator=(ALParseFile&&) = delete;
+
+    ~ALParseFile()
+    {
+        if (!pathName.empty() && !keepFile)
+        {
+            lg2::debug("Removing {FILE}", "FILE", pathName);
+            std::filesystem::remove(pathName);
+        }
+    }
+
+    /** @brief Create empty temporary file
+     */
+    ALParseFile()
+    {
+        std::string tempFile = std::filesystem::temp_directory_path() /
+                               "auditLogJson-XXXXXX";
+
+        lg2::debug("Constructing ALParseFile template={NAME}", "NAME",
+                   tempFile);
+
+        int fd = mkstemp(tempFile.data());
+        if (fd == -1)
+        {
+            throw sdbusplus::xyz::openbmc_project::Common::File::Error::Open();
+        }
+
+        // Store path to temporary file
+        pathName = tempFile;
+
+        // Close file descriptor
+        if (close(fd) == -1)
+        {
+            // Delete temporary file.  The destructor won't be called because
+            // the exception below causes this constructor to exit without
+            // completing.
+            std::filesystem::remove(pathName);
+            throw sdbusplus::xyz::openbmc_project::Common::File::Error::Open();
+        }
+    }
+
+    /** @brief Create parsed file that is not removed on destruction
+     * @details Used for debug only
+     */
+    ALParseFile(std::string filePath)
```

Comment:
std::string_view would be fine here, or const std::string&

Reply:
Good point. Reading up on string_view seems like that is preferred for performance reasons. I'll use that one here and a couple other places as well.